pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -585,6 +598,115 @@ func (grb *globalRoleBindingLifecycle) grantRestrictedAdminUserClusterPermission
 	return returnErr
 }
 
+// reconcileNamespacedRoleBindings ensures that RoleBindings exist for each namespace listed in NamespacedRules
+// from the associated GlobalRole
+func (grb *globalRoleBindingLifecycle) reconcileNamespacedRoleBindings(globalRoleBinding *v3.GlobalRoleBinding) error {
+	var returnError error
+	gr, err := grb.grLister.Get("", globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole %s: %w", globalRoleBinding.GlobalRoleName, err)
+	}
+
+	roleBindingUIDs := []types.UID{}
+
+	for ns := range gr.NamespacedRules {
+		_, err := grb.nsLister.Get("", ns)
+		if err != nil {
+			returnError = multierror.Append(returnError, errors.Wrapf(err, "couldn't get namespace %s", ns))
+			continue
+		}
+
+		roleName := gr.Name + "-" + ns
+		roleRef := v1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     roleName,
+		}
+
+		subject := rbac.GetGRBSubject(globalRoleBinding)
+		rbName := globalRoleBinding.Name + "-" + ns
```

Comment:
Similar comment here about the use of `SafeConcatName` for the name of the binding - I don't think that this is dangerous for this particular type, but better safe than sorry.